File: CreditCardValidator/Javascript/CreditCardValidator.js
function getCardNumberArray (creditCardNumber) {
    
    const cardDigits = creditCardNumber.split('');
    const cardNumber = [];

    let index = 0;
    let intValue = 0;

    for (let digit of cardDigits) {
        intValue = digit - '0';
        cardNumber[index] = intValue;
        index++;
    }

    return cardNumber;

}

function getCardType (cardNumber) {

    let cardType = "";
        
    if(cardNumber[0] === 4) cardType = "Visa Card";
    else if(cardNumber[0] === 5) cardType = "MasterCard";
    else if(cardNumber[0] === 6) cardType = "Discover Card";
    else if(cardNumber[0] === 3 && cardNumber[1] === 7) cardType = "American Express Card";
    else cardType = "invalid card type";

    return cardType;
        
}


function doubleEverySecondDigit (cardNumber) {

    let count = 0;
    let sum = 0;

    for(let index = cardNumber.length - 1; index >= 0; index--) {

        count++;
        if (count % 2 === 0) sum += (cardNumber[index] * 2);
    }

    return sum;
}


function addEveryDigitInOddPlaces (cardNumber) {

    let count = 0;
    let sum = 0;

    for(let index = cardNumber.length - 1; index >= 0; index--) {

        count++;
        if (count % 2 != 0) sum += cardNumber[index];
    }

    return sum;
}

    
function validateCardNumber (cardNumber) {

    let validityStatus = "";

    let cardNumberLength = cardNumber.length;
    let sumOfDoubledDigits = doubleEverySecondDigit (cardNumber);
    let sumOfDigitsInOddPlaces = addEveryDigitInOddPlaces (cardNumber);

    let sum = sumOfDoubledDigits + sumOfDigitsInOddPlaces;

    if(cardNumberLength < 13 || cardNumberLength > 16) validityStatus = "Invalid";

    if(sum % 10 === 0) validityStatus = "Valid";
    else validityStatus = "Invalid";

    return validityStatus;
}
            

const prompt = require("prompt-sync")();

console.log("\nHello, kindly enter card number to verify");
let creditCardNumber = String(prompt(""));

const cardNumberArray = getCardNumberArray (creditCardNumber);

console.log("\n***************************************************\n");
console.log("**Credit Card Type: " + getCardType (cardNumberArray));
console.log("\n**Credit Card Number: " + creditCardNumber);
console.log("\n**Credit Card Digit Length: " + cardNumberArray.length);
console.log("\n**Credit Card Validity Status: " + validateCardNumber (cardNumberArray));
console.log("\n***************************************************\n");

File: CreditCardValidator/Java/CreditCardValidator.java
import java.util.Scanner;

public class CreditCardValidator {

    public static int [] getCardNumberArray (String creditCardNumber) {
    
        char [] cardDigits = creditCardNumber.toCharArray();
        int [] cardNumber = new int [cardDigits.length];

        int index = 0;
        int intValue = 0;

        for (char digit : cardDigits) {
            intValue = digit - '0';
            cardNumber[index] = intValue;
            index++;
        }

        return cardNumber;

    }

    public static String getCardType (int [] cardNumber) {

        String cardType = "";
        
        if(cardNumber[0] == 4) cardType = "Visa Card";
        else if(cardNumber[0] == 5) cardType = "MasterCard";
        else if(cardNumber[0] == 6) cardType = "Discover Card";
        else if(cardNumber[0] == 3 && cardNumber[1] == 7) cardType = "American Express Card";
        else cardType = "invalid card type";

        return cardType;
        
    }


    public static int doubleEverySecondDigit (int [] cardNumber) {

        int count = 0;
        int sum = 0;

        for(int index = cardNumber.length - 1; index >= 0; index--) {

            count++;
            if (count % 2 == 0) sum += (cardNumber[index] * 2);
        }

        return sum;
    }


    public static int addEveryDigitInOddPlaces (int [] cardNumber) {

        int count = 0;
        int sum = 0;

        for(int index = cardNumber.length - 1; index >= 0; index--) {

            count++;
            if (count % 2 != 0) sum += cardNumber[index];
        }

        return sum;
    }

    
    public static String validateCardNumber (int [] cardNumber) {

        String validityStatus = "";
        int cardNumberLength = cardNumber.length;

        int sumOfDoubledDigits = doubleEverySecondDigit (cardNumber);
        int sumOfDigitsInOddPlaces = addEveryDigitInOddPlaces (cardNumber);

        int sum = sumOfDoubledDigits + sumOfDigitsInOddPlaces;

        if(cardNumberLength < 13 || cardNumberLength > 16) validityStatus = "Invalid";

        if(sum % 10 == 0) validityStatus = "Valid";
        else validityStatus = "Invalid";

        return validityStatus;
    }
            

    public static void main(String[] args) {

        Scanner userInput = new Scanner(System.in);

        System.out.println("\nHello, kindly enter card number to verify");
        String creditCardNumber = userInput.nextLine();

        int [] cardNumberArray = getCardNumberArray (creditCardNumber);

        System.out.println("\n***************************************************\n");
        System.out.println("**Credit Card Type: " + getCardType (cardNumberArray));
        System.out.println("\n**Credit Card Number: " + creditCardNumber);
        System.out.println("\n**Credit Card Digit Length: " + cardNumberArray.length);
        System.out.println("\n**Credit Card Validity Status: " + validateCardNumber (cardNumberArray));
        System.out.println("\n***************************************************\n");

    }

}
